b2sdk/scan/folder.py

```diff
@@ -178,15 +178,43 @@ def ensure_non_empty(self):
             raise EmptyDirectory(self.root)
 
     def _walk_relative_paths(
-        self, local_dir: str, relative_dir_path: str, reporter,
-        policies_manager: ScanPoliciesManager
+        self,
+        local_dir: str,
+        relative_dir_path: str,
+        reporter,
+        policies_manager: ScanPoliciesManager,
+        visited_symlinks=None,
+        starting_dir: str = None,
     ):
         """
         Yield a File object for each of the files anywhere under this folder, in the
         order they would appear in B2, unless the path is excluded by policies manager.
 
         :param relative_dir_path: the path of this dir relative to the scan point, or '' if at scan point
         """
+
+        def is_ancestor_directory(potential_ancestor, dir):
```

Comment:
@ppolewicz  _walk_relative_paths uses string patha and this is why I decided to use string path arguments for this helper function. Do you recommend to use Path objects here? in that case this function simply becomes:

`
return potential_ancestor in potential_descendant.parents
`

so probably not worth to do it as a separate function. At the same time, converting string paths to Path objects seems clumsy and inconsistent. I would prefer do refactor _walk_relative_paths to use Path objects instead of path strings. What do you think?

Reply:
Since `_walk_relative_paths` is private, you should refactor it to use path. Generally pathlib is in the standard library now and we should use it when appropriate (which usually means we have to move away from string paths - and that's good)